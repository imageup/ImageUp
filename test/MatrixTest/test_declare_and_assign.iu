function main() -> int
{
	matrix : arr(10, 3);
	int : i;
	int : j;
	for(i = 0; i < RowLen(arr); i=i+1)
	{
		for(j = 0; j < ColLen(arr); j=j+1)
		{
			arr[i][j] = i*j;
		}
	}

	string : out;
	for(i = 0; i < RowLen(arr); i=i+1)
	{
		out = "row " + StrParse(i) + ": ";
		for(j = 0; j < ColLen(arr); j=j+1)
		{
			arr[i][j] = i*j;
			out = out + " " + StrParsef(arr[i][j]);
		}
		prints(out);
	}

	return 0;
}