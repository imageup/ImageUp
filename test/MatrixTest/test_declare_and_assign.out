row 0:  0.00000 0.00000 0.00000
row 1:  0.00000 1.00000 2.00000
row 2:  0.00000 2.00000 4.00000
row 3:  0.00000 3.00000 6.00000
row 4:  0.00000 4.00000 8.00000
row 5:  0.00000 5.00000 10.0000
row 6:  0.00000 6.00000 12.0000
row 7:  0.00000 7.00000 14.0000
row 8:  0.00000 8.00000 16.0000
row 9:  0.00000 9.00000 18.0000
